### PR TITLE
Allow enabling debug logs

### PR DESCRIPTION
**Pull request by HoussemNasri**
Just pass `--debug` to the command line.

image

### Mandatory checks
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/cli/Launcher.java [hunk 1]
```diff
@@ -43,9 +44,15 @@ public class Launcher {
...
     private static String[] ARGUMENTS;
 
+    private static boolean isDebugEnabled = false;
+
     public static void main(String[] args) {
         ARGUMENTS = args;
 
+        if (Arrays.stream(ARGUMENTS).anyMatch("--debug"::equalsIgnoreCase)) {
+            isDebugEnabled = true;
+        }
```
Comment (koppor) [inline src/main/java/org/jabref/cli/Launcher.java:54]:
```suggestion
        JabRefCLI = cli = new org.jabref.cli.JabRefCLI(args);
        isDebugEnabled = cli.isDebugLogging();
```

File: src/main/java/org/jabref/cli/Launcher.java [hunk 2]
```diff
@@ -43,9 +44,15 @@ public class Launcher {
     private static Logger LOGGER;
     private static String[] ARGUMENTS;
 
+    private static boolean isDebugEnabled = false;
```
Comment (koppor) [inline src/main/java/org/jabref/cli/Launcher.java:47]:
```suggestion
    private static boolean isDebugEnabled;
```

**Review by koppor — CHANGES_REQUESTED** [review]
The idea is good. I saw, there already is a --debug switch:

https://github.com/JabRef/jabref/blob/6c680ea7cc8b43a22c155444831fe4735040bf61/src/main/java/org/jabref/cli/ArgumentProcessor.java#L190-L190

The check for --debug has to be done very early (due to our "broken" argument handling).

Suggestion (because of code reuse)

- Remove these three lines (the if statement)
- Use JabRef CLI directly to parse the arguemnts and handle it.

File: src/main/java/org/jabref/cli/Launcher.java (hunk 1 above)
Comment (HoussemNasri) [reply src/main/java/org/jabref/cli/Launcher.java]: Thanks! JabRefCLI throws an exception so I wrapped it in a try-catch statement.

**Review by Siedlerchr — DISMISSED** [review]

**Comment by Siedlerchr** [comment]
we need to test the binary that comes out of this if it works

**Comment by HoussemNasri** [comment]
I tested it locally by creating an image with jlink, and the logs are written correctly to the selected log file.

**Review by koppor — APPROVED** [review]

**Comment by Siedlerchr** [comment]
xception in thread "main" java.lang.UnsupportedOperationException: Configuration cannot be changed after applying to tinylog
        at org.tinylog.api@2.6.2/org.tinylog.configuration.Configuration.set(Configuration.java:288)
        at org.jabref@100.0.0/org.jabref.cli.Launcher.lambda$addLogToDisk$0(Launcher.java:124)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at org.jabref@100.0.0/org.jabref.cli.Launcher.addLogToDisk(Launcher.java:124)
        at org.jabref@100.0.0/org.jabref.cli.Launcher.main(Launcher.java:60)

**Comment by Siedlerchr** [comment]
Found the issue. It was the Localization.lang call

**Comment by HoussemNasri** [comment]
Oops, I didn't test it after using JabRefCLI

**Pull request merged**